cpmpy/tools/explain/mss.py

```diff
@@ -82,13 +87,17 @@ def mss_grow_naive(soft, hard=[], solver="ortools"):
 
     to_check = list(soft)
     sat_subset = []
+    s = cp.SolverLookup.get(solver)
+    s += hard
 
     while len(to_check):
         c = to_check.pop()
-        if cp.Model(sat_subset + [c] + hard).solve(solver=solver) is True:
-            sat_subset += [c]
+        s += c
+        if s.solve() is True:
+            sat_subset.append(c)
         else:
-            # UNSAT, cannot add to sat subset
-            pass
+            # UNSAT, cannot add to sat subset, reset solver to just sat subset
+            s = cp.SolverLookup.get(solver)
+            s += hard
```

Comment:
sorry to re-open, but should be hard + sat_subset?

Reply:
Yes, you're right, I'll push the fix to main - pretty surprising the tests did not catch this...